src/helpers/amazon-q/amazon-q-client.ts

```diff
@@ -45,21 +47,42 @@ export const callClient = async (
   attachments: AttachmentInput[],
   env: SlackEventsEnv,
   iamSessionCreds: Credentials,
+  commandType: 'sync' | 'stream',
   context?: {
     conversationId: string;
     parentMessageId: string;
   }
-): Promise<ChatSyncCommandOutput> => {
-  const input = {
+): Promise<ChatSyncCommandOutput | ChatCommandOutput> => {
+  let input: any = {
```

Comment:
nit: usually we want to avoid "any" because it sometimes automatically gets flagged. I think you could just put the input init inside each if else block

Reply:
Also, generally we try to avoid conditional logic (here we have conditionals for commandType and also input/output types. You could try moving the conditional one level higher and create two separate callClient APIs (btw I feel this method name was not clear in the first place to what it does)

Reply:
ack. Will change to 2 functions instead.